flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractInput.java

```diff
@@ -35,7 +38,8 @@
  * AbstractStreamOperatorV2}.
  */
 @Experimental
-public abstract class AbstractInput<IN, OUT> implements Input<IN>, KeyContextHandler {
+public abstract class AbstractInput<IN, OUT>
```

Comment:
It looks like only some test classes inherit `AbstractInput`, I'm worried it won't really work.

Reply:
You're right. The `AbstractInput` is only for user to extend and implement their own input. The proper way is to provide such logic around `InputBase`.